beacon-chain/sync/blobs_test.go

```diff
@@ -103,32 +107,45 @@ func generateTestBlockWithSidecars(t *testing.T, parent [32]byte, slot types.Slo
 	root, err := block.Block.HashTreeRoot()
 	require.NoError(t, err)
 
-	sidecars := make([]*ethpb.DeprecatedBlobSidecar, len(commitments))
+	sbb, err := blocks.NewSignedBeaconBlock(block)
+	require.NoError(t, err)
+	sidecars := make([]blocks.ROBlob, len(commitments))
 	for i, c := range block.Block.Body.BlobKzgCommitments {
-		sidecars[i] = generateTestSidecar(root, block, i, c)
+		sidecars[i] = generateTestSidecar(t, root, sbb, i, c)
 	}
 	return block, sidecars
 }
 
-func generateTestSidecar(root [32]byte, block *ethpb.SignedBeaconBlockDeneb, index int, commitment []byte) *ethpb.DeprecatedBlobSidecar {
+func generateTestSidecar(t *testing.T, root [32]byte, block interfaces.ReadOnlySignedBeaconBlock, index int, commitment []byte) blocks.ROBlob {
+	header, err := interfaces.SignedBeaconBlockHeaderFromBlockInterface(block)
```

Comment:
I think this is cleaner

```suggestion
	header, err := block.Header()
```

Reply:
oh nice, I missed that we have that helper!

Reply:
https://github.com/prysmaticlabs/prysm/pull/13190/commits/98e5278f323794e2862cbbe54c40d8da4c0c1154